integration-tests/jpa-postgresql-vault/src/main/java/io/quarkus/it/jpa/postgresql/JPAFunctionalityTestEndpoint.java

```diff
@@ -0,0 +1,167 @@
+    EntityManagerFactory entityManagerFactory;
+
+    @ConfigProperty(name = "password")
+    String someSecret;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        try {
+            log.info("someSecret={}", someSecret);
+            doStuffWithHibernate(entityManagerFactory);
+        } catch (Exception e) {
+            reportException("An error occurred while performing Hibernate operations", e, resp);
+        }
+        resp.getWriter().write("OK");
+    }
+
+    /**
+     * Lists the various operations we want to test for:
+     */
+    private static void doStuffWithHibernate(EntityManagerFactory entityManagerFactory) {
+
+        //Cleanup any existing data:
+        deleteAllPerson(entityManagerFactory);
+
+        //Store some well known Person instances we can then test on:
+        storeTestPersons(entityManagerFactory);
+
+        //Load all persons and run some checks on the query results:
+        verifyListOfExistingPersons(entityManagerFactory);
+
+        //Try a JPA named query:
+        verifyJPANamedQuery(entityManagerFactory);
+
+        deleteAllPerson(entityManagerFactory);
+
+    }
+
+    private static void verifyJPANamedQuery(final EntityManagerFactory emf) {
+        EntityManager em = emf.createEntityManager();
+        EntityTransaction transaction = em.getTransaction();
+        transaction.begin();
+        TypedQuery<Person> typedQuery = em.createNamedQuery(
+                "get_person_by_name", Person.class);
+        typedQuery.setParameter("name", "Quarkus");
+        final Person singleResult = typedQuery.getSingleResult();
+
+        if (!singleResult.getName().equals("Quarkus")) {
+            throw new RuntimeException("Wrong result from named JPA query");
+        }
+
+        transaction.commit();
+        em.close();
+    }
+
+    private static void verifyListOfExistingPersons(final EntityManagerFactory emf) {
+        EntityManager em = emf.createEntityManager();
+        EntityTransaction transaction = em.getTransaction();
+        transaction.begin();
+        listExistingPersons(em);
+        transaction.commit();
+        em.close();
+    }
+
+    private static void storeTestPersons(final EntityManagerFactory emf) {
+        EntityManager em = emf.createEntityManager();
+        EntityTransaction transaction = em.getTransaction();
+        transaction.begin();
+        persistNewPerson(em, "Gizmo");
+        persistNewPerson(em, "Quarkus");
+        persistNewPerson(em, "Hibernate ORM");
+        transaction.commit();
+        em.close();
+    }
+
+    private static void deleteAllPerson(final EntityManagerFactory emf) {
+        EntityManager em = emf.createEntityManager();
+        EntityTransaction transaction = em.getTransaction();
+        transaction.begin();
+        em.createNativeQuery("Delete from Person").executeUpdate();
+        transaction.commit();
+        em.close();
```

Comment:
We really don't need all that. We just need to check that we can do a simple access to the db via Hibernate ORM. Let's not clutter the tests with unrelated things.

Reply:
I simplified it a lot. Initially I blindly copy/pasted the jpa-postgresql module because I knew it was working and it was doing a lots of things (such as reflection), and so it was easy to see if vault's integration brought any regression.
But now I am much more comfortable in vault it being a more focused test.